fds.go

```diff
@@ -135,8 +160,9 @@ func (f *Fds) listenerLocked(network, addr string) (net.Listener, error) {
 
 // AddListener adds a listener.
 //
-// It is safe to close ln after calling the method.
-// Any existing listener with the same address is overwitten.
+// It remains the caller's responsibility to close the passed listener; only a
+// duplicated file descriptor is stored in Fds.
+// Any existing added listener with the same address is overwitten.
```

Comment:
I think the old fd is closed because `os.File` in go has a finalizer that does a close to avoid fd leaks. I don't think this library's use of `fcntl(F_DUPFD)` is enough to avoid the finalizer because it does pass that fd through to `os.NewFile` right after.

As such, I think the risk is just that it's overwritten, not that it's leaked.

Reply:
Fair enough. That is still *technically* a leak, though, even if in practice it won't happen because GC will eventually get it. Given the right conditions, the GC could not run for an arbitrarily long time.